### ISSUE TITLE
core/odin/parser segmentation fault on linux

### ISSUE THREAD

**Issue by DanielGavin** [odin-lang/Odin#981]
## Context

Commit d353f97f91c18fd910c4c16604dcaf3e3afbc5f2 introduced a new bug in the odin/parser on Linux/MacOS.

If you uncomment this line again it works:  https://github.com/odin-lang/Odin/blob/63b54ce7c695fb1e204fe77db029cdebf7206b28/src/llvm_backend.cpp#L2555

* Operating System:
Manjaro Linux

Lastest odin commit be12f12c3c2872fd89021af16c2ab838277c4c4a

### Code

``` odin
import "core:odin/ast"
import "core:odin/parser"

main :: proc() {
    p := parser.default_parser();

    pkg := new(ast.Package);
    pkg.kind     = .Normal;
    pkg.fullpath = "";

    src := `package test
        main :: proc() {
            my_slice := make_slice([]My_Struct, 23);
            my_slic
        }
    `;

    ast := ast.File {
        fullpath = "",
        src = transmute([]u8)src,
        pkg = pkg,
    };

    parser.parse_file(&p, &ast);
}
```

## Expected Behavior

Parses the file and prints out syntax error.

## Current Behavior

Gives a segmentation fault when it prints the syntax error.

**Comment by gingerBill** [comment]
Is this a bug on both Linux and MacOS, or just Linux?

**Comment by DanielGavin** [comment]
I can't verify if it's the same error on MacOS(Don't own one), only that it gives the exact same error when run with CI on MacOS as Linux.

**Comment by DanielGavin** [comment]
It is has now been fixed since the last commit b8f8d4c3a12ea85285511112b33425923418556a.
